### PR TITLE
Conditional map support

### PR DESCRIPTION
**Pull request by moonheart08**
## About the PR 
Maps can now have conditions required for them to be votable
That is all
```yaml
  conditions:
  - !type:HolidayMapCondition
    holidays: [ "Christmas" ]
```

### CODE REVIEW

File: Content.Server/Maps/GameMapManager.cs [hunk 1]
```diff
@@ -111,7 +111,9 @@ public bool CheckMapExists(string gameMap)
 
     private bool IsMapEligible(GameMapPrototype map)
     {
-        return map.MaxPlayers >= _playerManager.PlayerCount && map.MinPlayers <= _playerManager.PlayerCount;
+        return map.MaxPlayers >= _playerManager.PlayerCount &&
+               map.MinPlayers <= _playerManager.PlayerCount &&
```
Comment (Efruit) [inline Content.Server/Maps/GameMapManager.cs:115]: Shouldn't the player levels be made in to a `GameMapCondition`?

File: Content.Server/Maps/Conditions/HolidayMapCondition.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+public class HolidayMapCondition : GameMapCondition
+{
+    [DataField("holidays")]
+    public string[] Holidays { get; } = default!;
+
+    public override bool Check(GameMapPrototype map)
+    {
+        var holidaySystem = EntitySystem.Get<HolidaySystem>();
+        Logger.Debug($"{Holidays.Any(holiday => holidaySystem.IsCurrentlyHoliday(holiday)) ^ Inverted}");
```
Comment (Efruit) [inline Content.Server/Maps/Conditions/HolidayMapCondition.cs:19]: d e b u g
```suggestion
```

File: Content.Server/Maps/GameMapManager.cs (hunk 1 above)
Comment (moonheart08) [reply Content.Server/Maps/GameMapManager.cs:115]: Opted against it as EVERY MAP has player min/max, so it'd just be extra typing for no gain.

**Review by Efruit — CHANGES_REQUESTED** [review]
99% good
just needs a single de-debugging (and maybe a followup that moves player count to its own `GameMapCondition`)

not that my reviews count for shit, but  ¯\\\_(シ)\_/¯

File: Content.Server/Maps/GameMapManager.cs (hunk 1 above)
Comment (Efruit) [reply Content.Server/Maps/GameMapManager.cs:115]: dunno how universal the player count restriction will be, especially if servers have a steady general population level, and most of their maps cater thereto

anyway :thumbsup:

Comment (PaulRitter) [reply Content.Server/Maps/GameMapManager.cs:115]: i agree with efr that playercount should p be a condition and not hardbaked into the gamemapprototype, but i wont force you to change it in this pr.

**Review by PaulRitter — APPROVED** [review]

**Review by Zumorica — APPROVED** [review]

**Pull request merged**